src/main/kotlin/no/nav/syfo/behandler/database/BehandlerDialogmeldingQuery.kt

```diff
@@ -0,0 +1,150 @@
+    personIdentNumber: PersonIdentNumber,
+    behandlerDialogmeldingId: Int,
+) {
+    val uuid = UUID.randomUUID()
+    val idList = this.prepareStatement(queryCreateBehandlerDialogmeldingArbeidstaker).use {
+        it.setString(1, uuid.toString())
+        it.setString(2, personIdentNumber.value)
+        it.setTimestamp(3, Timestamp.from(Instant.now()))
+        it.setInt(4, behandlerDialogmeldingId)
+        it.executeQuery().toList { getInt("id") }
+    }
+
+    if (idList.size != 1) {
+        throw SQLException("Creating BehandlerDialogmeldingArbeidstaker failed, no rows affected.")
+    }
+}
+
+fun Connection.createBehandlerDialogmelding(
+    newBehandler: NewBehandler,
+): Pair<Int, UUID> {
+    val behandlerRef = UUID.randomUUID()
+    val now = Timestamp.from(Instant.now())
+    val behandlerDialogmeldingIdList = this.prepareStatement(queryCreateBehandlerDialogmelding).use {
+        it.setString(1, behandlerRef.toString())
+        it.setString(2, newBehandler.type.name)
+        it.setString(3, newBehandler.personident?.value)
+        it.setString(4, newBehandler.fornavn)
+        it.setString(5, newBehandler.mellomnavn)
+        it.setString(6, newBehandler.etternavn)
+        it.setString(7, newBehandler.partnerId)
+        it.setString(8, newBehandler.herId)
+        it.setString(9, newBehandler.parentHerId)
+        it.setString(10, newBehandler.hprId)
+        it.setString(11, newBehandler.kontor)
+        it.setString(12, newBehandler.adresse)
+        it.setString(13, newBehandler.postnummer)
+        it.setString(14, newBehandler.poststed)
+        it.setString(15, newBehandler.orgnummer?.value)
+        it.setString(16, newBehandler.telefon)
+        it.setTimestamp(17, now)
+        it.setTimestamp(18, now)
+        it.executeQuery().toList { getInt("id") }
+    }
+
+    if (behandlerDialogmeldingIdList.size != 1) {
+        throw SQLException("Creating BehandlerDialogmelding failed, no rows affected.")
+    }
+
+    return Pair(behandlerDialogmeldingIdList.first(), behandlerRef)
+}
+
+const val queryCreateBehandlerDialogmelding =
+    """
+        INSERT INTO BEHANDLER_DIALOGMELDING (
+            id,
+            behandler_ref,
+            type,
+            personident,
+            fornavn,
+            mellomnavn,
+            etternavn,
+            partner_id,
+            her_id,
+            parent_her_id,
+            hpr_id,
+            kontor,
+            adresse,
+            postnummer,
+            poststed,
+            orgnummer,
+            telefon,
+            created_at,
+            updated_at
+            ) VALUES (DEFAULT, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) RETURNING id
+    """
+
+const val queryGetBehandlerDialogmeldingForArbeidstaker =
+    """
+        SELECT BEHANDLER_DIALOGMELDING_ARBEIDSTAKER.arbeidstaker_personident, BEHANDLER_DIALOGMELDING.*
+        FROM
+        BEHANDLER_DIALOGMELDING
+        LEFT JOIN BEHANDLER_DIALOGMELDING_ARBEIDSTAKER 
```

Comment:
left join  sammen med betingelsen ellers betyr så vidt jeg forstår at du får alle radene i behandler_dialogmelding i resultsettet. Hadde det ikke vært bra å ha partner_id som en betingelse i tillegg?

Reply:
Ja, riktig, jeg ser det nå. Hadde partner_id som en betingelse til å begynne med, men fjernet den (husker ikke helt hvorfor 😅 ). Bør nok endres. Kanskje er det ryddigere å ha én spørring som bare sjekker mot behandler_dialogmelding på partnerId og en ny som sjekker om en behandler_dialogmelding er knyttet til personident 🤔

Reply:
Med inner join vil man vel ikke få ut behandlere som finnes, men som er knyttet til en annen personident (dvs de hvor vi bare skal opprette kobling)? Sånn sett er det vel greiest å dele opp i to spørringer.

Reply:
Fikset!